pkg/kubelet/prober/prober_manager.go

```diff
@@ -247,16 +247,16 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 
 		if started {
 			var ready bool
-			if c.State.Running == nil {
-				ready = false
-			} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok {
+			if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok {
```

Comment:
just noting:

@thockin earlier commit: https://github.com/kubernetes/kubernetes/commit/64d42542174d22968c424ba4b976b5e1055647cd

started will be `false` above if c.State.Running is nil based on above check, so checking for nil again here was redundant.

Reply:
Sorry for replay too later，
When the kubelet is restarted, the container with the readness probe will be temporarily unhealthy. To solve this problem, another code is modified. When the kubelet is restarted, the readness probe status is retained. Therefore, the value of started may not be false in our code.
I will close this issue and create a new issue to solve the problem that the container that is running properly is temporarily unhealthy after the kubelet is restarted.